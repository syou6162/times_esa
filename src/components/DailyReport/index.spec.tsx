import { fireEvent, render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { DailyReport, DailyReportProps } from '.'

describe('DailyReportが正しく表示される', () => {
  it('ディフォルトではhtmlが表示される', () => {
    const props: DailyReportProps = {
      fetching: false,
      fetchErrorMessage: "",

      esaText: "hello world!",
      esaHtml: "<span>HTML TEXT<span>",
      reloadDailyReport: () => { }
    }
    const renderResult = render(
      <DailyReport {...props} />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
    expect(renderResult.getByText("HTML TEXT")).toBeTruthy();
    expect(renderResult.queryByText("hello world")).toBeNull();
  });

  it('textをクリックするとtextが表示される', async () => {
    const props: DailyReportProps = {
      fetching: false,
      fetchErrorMessage: "",

      esaText: "RAW TEXT",
      esaHtml: "<span>hello world!<span>",
      reloadDailyReport: () => { }
    }
    const renderResult = render(
      <DailyReport {...props} />
    );

    fireEvent.click(renderResult.getByText("text"));
    await waitFor(() => {
      expect(renderResult.asFragment()).toMatchSnapshot();
      expect(renderResult.getByText("RAW TEXT")).toBeTruthy();
      expect(renderResult.queryByText("hello world")).toBeNull();
    });
  });
  
  it('shareをクリックすると共有用の画面が表示される', async () => {
    const props: DailyReportProps = {
      fetching: false,
      fetchErrorMessage: "",

      esaText: "12:34 RAW TEXT\n\n---\n\n13:15 hoge\n\n---",
      esaHtml: "<span>hello world!<span>",
      reloadDailyReport: () => { }
    }
    const renderResult = render(
      <DailyReport {...props} />
    );

    fireEvent.click(renderResult.getByText("share"));
    await waitFor(() => {
      expect(renderResult.asFragment()).toMatchSnapshot();
      expect(renderResult.getAllByText("コピーする")).toHaveLength(2);
      expect(renderResult.queryByText("hello world")).toBeNull();
    });
  });
}); 