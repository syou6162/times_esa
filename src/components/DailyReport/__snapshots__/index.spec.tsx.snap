// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyReportが正しく表示される shareをクリックすると共有用の画面が表示される 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      Update
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      html
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      text
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      share
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div>
      <div>
        <div>
          <div
            style="white-space: pre-wrap; word-wrap: break-word; text-align: left; justify-content: left; align-items: left;"
          >
            12:34 RAW TEXT
          </div>
          <div>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              href="https://twitter.com/intent/tweet?text=RAW%20TEXT"
              style="margin: 5px; text-transform: none; float: right;"
              tabindex="0"
            >
              Tweetする
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              style="margin: 5px; text-transform: none; float: right;"
              tabindex="0"
              type="button"
            >
              コピーする
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <hr
            style="clear: both;"
          />
        </div>
        <div>
          <div
            style="white-space: pre-wrap; word-wrap: break-word; text-align: left; justify-content: left; align-items: left;"
          >
            13:15 hoge
          </div>
          <div>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              href="https://twitter.com/intent/tweet?text=hoge"
              style="margin: 5px; text-transform: none; float: right;"
              tabindex="0"
            >
              Tweetする
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              style="margin: 5px; text-transform: none; float: right;"
              tabindex="0"
              type="button"
            >
              コピーする
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <hr
            style="clear: both;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DailyReportが正しく表示される textをクリックするとtextが表示される 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      Update
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      html
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      text
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      share
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div>
      <div
        style="white-space: pre-wrap; word-wrap: break-word; text-align: left; justify-content: left; align-items: left;"
      >
        RAW TEXT
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DailyReportが正しく表示される ディフォルトではhtmlが表示される 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      Update
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      html
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      text
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-9aa3p4-MuiButtonBase-root-MuiButton-root"
      style="margin: 3px; text-transform: none;"
      tabindex="0"
      type="button"
    >
      share
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div>
      <div
        style="white-space: normal; overflow: hidden; text-align: left; justify-content: left; align-items: left;"
      >
        <span>
          HTML TEXT
          <span />
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
